app/lib/zap/controller.js

```diff
@@ -149,20 +151,16 @@ class ZapController {
     this._registerIpcListeners()
 
     // Disconnect from any existing lightning wallet connection.
-    if (lifecycle.from === 'connected') {
-      if (this.lightning && this.lightning.can('disconnect')) {
-        this.lightning.disconnect()
-      }
-      if (this.walletUnlocker && this.walletUnlocker.can('disconnect')) {
-        this.walletUnlocker.disconnect()
-      }
-      this.sendMessage('lndStopped')
+    if (this.lightning && this.lightning.can('disconnect')) {
+      this.lightning.disconnect()
     }
-
-    // If we are comming from a running state, stop the Neutrino process.
-    else if (lifecycle.from === 'running') {
-      await this.shutdownNeutrino()
+    if (this.walletUnlocker && this.walletUnlocker.can('disconnect')) {
+      this.walletUnlocker.disconnect()
     }
+    this.sendMessage('lndStopped')
+
+    // Shut down Neutrino.
+    await this.shutdownNeutrino()
 
     // Give the grpc connections a chance to be properly closed out.
     await new Promise(resolve => setTimeout(resolve, 200))
```

Comment:
consider using 
`https://github.com/LN-Zap/zap-desktop/blob/master/app/lib/utils/promiseTimeout.js`

Reply:
I'll raise a separate PR for refactoring that.